### PR TITLE
implement error handling

### PR DESCRIPTION
**Pull request by Julia-Protasevich**

### CODE REVIEW

**Comment by Julia-Protasevich** [comment]
I don't pretend like user controller methods are working, I simply removed all mentions of "context", so now they look more ExpressJS-ish.

File: middleware/error_handler.js [hunk 1]
```diff
@@ -0,0 +1,6 @@
+import logger from '../config/logger/logger-service.js';
+
+export default (err, req, res, next) => {
+    res.status(500).send('Oops!');
```
Comment (MihailMartinovich) [inline middleware/error_handler.js:4]: what will get the client if the error happened somewhere was 403/401?

File: middleware/error_handler.js [hunk 2]
```diff
@@ -0,0 +1,6 @@
...
+    logger.warn("Error in controller: ", err);
```
Comment (MihailMartinovich) [inline middleware/error_handler.js:5]: it's better to log error before sending a response.

File: controllers/users.controller.js [hunk 3]
```diff
@@ -4,10 +4,14 @@ import passport from '../authentication/passport.js';
 import logger from '../config/logger/logger-service.js';
 
 export const UsersController = {
-  async login(ctx, next){
+  async login(req, res, next){
     await passport.authenticate('local', function (err, user) {
+      if (err) {
+        return next(err);
```
Comment (MihailMartinovich) [inline controllers/users.controller.js:10]: why are you using in some placec "next(err)" and "return next(err)" in others?

File: controllers/books.controller.js [hunk 4]
```diff
@@ -1,50 +1,70 @@
...
+				isTakenByUser: req.query.userId
+			}, {
+				new: true,
+				returnNewDocument: true,
+			},
+			function (err, result) {
+				if (err) {
+					next(err);
+				} else if (result) {
+					res.status(200).send(result);
```
Comment (MihailMartinovich) [inline controllers/books.controller.js:45]: use in future res.json(). It sets correct content-type for your response.

File: middleware/error_handler.js (hunk 1 above)
Comment (Julia-Protasevich) [reply middleware/error_handler.js]: Added check if error had a previously set status

File: middleware/error_handler.js (hunk 2 above)
Comment (Julia-Protasevich) [reply middleware/error_handler.js:4]: switched lines

File: controllers/users.controller.js (hunk 3 above)
Comment (Julia-Protasevich) [reply controllers/users.controller.js:10]: Added return for all next()-s

File: controllers/books.controller.js (hunk 4 above)
Comment (Julia-Protasevich) [reply controllers/books.controller.js]: replaced for json

**Pull request merged**